.ci/static-checks.sh

```diff
@@ -536,14 +541,77 @@ check_docs()
 			git ls-remote "$url" > /dev/null 2>&1 && continue
 		fi
 
+		info "Checking URL $url"
+
 		# Check the URL, saving it if invalid
-		( curl -sLf -o /dev/null "$url" ||\
-				echo "$url" >> "$invalid_urls") &
+		#
+		# Each URL is checked in a separate process as each unique URL
+		# requires us to hit the network.
+		(
+			local ret
+
+			local curl_out=$(mktemp)
+
+			# Process specific file to avoid out-of-order writes
+			local invalid_file=$(printf "%s/%d" "$invalid_urls_dir" "$$")
+
+			{ curl -sIL --max-time "$url_check_timeout_secs" "$url" &>"$curl_out"; ret=$?; } || true
+
+			# A transitory error, or the URL is incorrect,
+			# but capture either way.
+			if [ "$ret" -ne 0 ]; then
+				echo "$url" >> "${invalid_file}"
+				continue
```

Comment:
@jodh-intel are you sure `continue` is handled properly inside `( )`?

Reply:
I've replaced those with `exit` calls now.